### PR TITLE
Beetmover prefixes rebased

### PR DESCRIPTION
**Pull request by hneiva**
Rebased #355

### CODE REVIEW

File: beetmoverscript/docker.d/init_worker.sh
Comment (MihaiTabara) [inline beetmoverscript/docker.d/init_worker.sh:82]: We should double-check that we export these values properly in relengworker SOPS for the production ones. Otherwise, if we don't, when we land this to the production workers, the GCP workers will barf.

File: beetmoverscript/docker.d/init_worker.sh
Comment (MihaiTabara) [inline beetmoverscript/docker.d/init_worker.sh:70]: We should double-check we export these values properly, even though failing GCP dev/staging workers are probably less of a problem.

File: beetmoverscript/docker.d/worker.yml
Comment (MihaiTabara) [inline beetmoverscript/docker.d/worker.yml]: it's nice that we've included focus too. we may need that at some point too in the future 👍

File: beetmoverscript/docker.d/worker.yml
Comment (MihaiTabara) [inline beetmoverscript/docker.d/worker.yml:186]: This whole section looks great!

File: beetmoverscript/src/beetmoverscript/constants.py
Comment (MihaiTabara) [inline beetmoverscript/src/beetmoverscript/constants.py]: I'm not very happy about the way I named this function since we're going to reuse this for Fenix nightlies too. But after all it's just naming thing since the logic is dictated in the task payload. Though we may want to revisit and s/direct-push-to-releases/direct-push-to-bucket` or alike so make sure it's not confusing us in the future.

File: beetmoverscript/src/beetmoverscript/script.py
Comment (MihaiTabara) [inline beetmoverscript/src/beetmoverscript/script.py]: We may want to rename this function to direct_push_to_bucket at some point since we're going to use it not only to push Fenix into ~releases direcrtly, but also in ~nightly buckets. Not blocking for now, but may help with confusion in the future.

**Review by MihaiTabara — APPROVED** [review]
This LGTM to me modulo two things:
* make sure we run successfully a nightly staging release - I have some concerns around the schema of the task payload that might not match. Also make sure the beetmover task scopes are working as expected
* confirm SOPS secrets are well defined so that we don't break the production workers

File: beetmoverscript/docker.d/init_worker.sh
Comment (escapewindow) [reply beetmoverscript/docker.d/init_worker.sh:82]: I think these are exported for all beetmovers: https://github.com/mozilla-services/cloudops-infra/blob/master/projects/relengworker/k8s/charts/beetmover/templates/configmap.yaml#L12-L24 and https://github.com/mozilla-services/cloudops-infra/blob/master/projects/relengworker/k8s/charts/beetmover/templates/secret.yaml#L12-L22

This also means we could test for these vars over both dep and prod, since we'll just have values like `"fake"` in dep/dev.

File: beetmoverscript/src/beetmoverscript/data/artifactMap_beetmover_task_schema.json
Comment (escapewindow) [inline beetmoverscript/src/beetmoverscript/data/artifactMap_beetmover_task_schema.json:60]: Am I right in thinking that these keys may be required in some types of behaviors but not others?
If so, we might want to validate that these keys exist for the behaviors that need them... we can do that with multiple jsonschema files (one per behavior?) or by just making sure the task is well-formed for each behavior in the python code.

File: beetmoverscript/src/beetmoverscript/task.py
Comment (escapewindow) [inline beetmoverscript/src/beetmoverscript/task.py:44]: We've moved away from action scopes elsewhere, to putting the behavior/action in the task payload, but that shouldn't block here.

**Review by escapewindow — COMMENTED** [review]
Other than wondering if changing the required schema looks like will affect other actions, this looks good, thank you!

File: beetmoverscript/src/beetmoverscript/task.py
Comment (hneiva) [reply beetmoverscript/src/beetmoverscript/task.py:44]: is it an easy migration? Worth doing here? We've got some time still

Comment (escapewindow) [reply beetmoverscript/src/beetmoverscript/task.py:44]: The steps would probably be:

- support setting the action/behavior from a task payload key/value pair
  - add that to the schema, look there first, fall back to the scope
  - roll that out to production beetmover
- change the in-tree configs to use the task payload instead
- repeat for all branches and repos that use the scope to set the action/behavior
- once all uses of action scopes are in the task payload instead of scope, remove the scope and support for reading the action from the scope

If we introduced the action scope in this PR, that would be pretty easy. I suspect we're using beetmover action scopes all over gecko and other repos, so the in-tree portion might be lengthy.

File: beetmoverscript/docker.d/worker.yml
Comment (ahal) [reply beetmoverscript/docker.d/worker.yml]: I think `focus` should be `focus-android` now?

File: beetmoverscript/docker.d/worker.yml
Comment (ahal) [inline beetmoverscript/docker.d/worker.yml:154]: This might need to be `focus-android` too? I really don't know though.

**Comment by hneiva** [comment]
@escapewindow can you think of any reason not to merge this in? We've been using it for the past week without issues and would like to start working towards prod archive

File: beetmoverscript/docker.d/worker.yml
Comment (escapewindow) [reply beetmoverscript/docker.d/worker.yml]: let's `s,focus,focus-android,` here before merging if possible, or quick followup?

File: beetmoverscript/docker.d/worker.yml
Comment (escapewindow) [reply beetmoverscript/docker.d/worker.yml:154]: yeah, let's maybe leave this til we know and we can adjust if/when we beetmove focus and find bustage.

File: beetmoverscript/src/beetmoverscript/data/artifactMap_beetmover_task_schema.json
Comment (escapewindow) [reply beetmoverscript/src/beetmoverscript/data/artifactMap_beetmover_task_schema.json:60]: let's file a followup on this.

File: beetmoverscript/src/beetmoverscript/task.py
Comment (escapewindow) [reply beetmoverscript/src/beetmoverscript/task.py:44]: let's punt on moving away from action scopes. sometime in the unplanned future.

**Review by escapewindow — CHANGES_REQUESTED** [review]
Looks like I just want the focus -> focus-android scope prefix change, then let's merge!

**Review by escapewindow — APPROVED** [review]

**Pull request merged**
